src/_ert/forward_model_runner/reporting/event.py

```diff
@@ -68,45 +66,48 @@ def __init__(self, evaluator_url, token=None, cert_path=None):
         else:
             self._cert = None
 
-        self._statemachine = StateMachine()
-        self._statemachine.add_handler((Init,), self._init_handler)
-        self._statemachine.add_handler((Start, Running, Exited), self._job_handler)
-        self._statemachine.add_handler((Checksum,), self._checksum_handler)
-        self._statemachine.add_handler((Finish,), self._finished_handler)
-
         self._ens_id = None
         self._real_id = None
-        self._event_queue: queue.Queue[events.Event | EventSentinel] = queue.Queue()
-        self._event_publisher_thread = ErtThread(target=self._event_publisher)
+        self._event_queue: asyncio.Queue[events.Event | EventSentinel] = asyncio.Queue()
+
         self._timeout_timestamp = None
-        self._timestamp_lock = threading.Lock()
         # seconds to timeout the reporter the thread after Finish() was received
         self._reporter_timeout = 60
 
-    def _event_publisher(self):
+        self._queue_polling_timeout = 2
+        self._event_publishing_task = asyncio.create_task(self.async_event_publisher())
+
+    async def join(self) -> None:
+        await self._event_publishing_task
+
+    async def async_event_publisher(self):
         logger.debug("Publishing event.")
-        with Client(
+        async with Client(
             url=self._evaluator_url,
             token=self._token,
             cert=self._cert,
         ) as client:
             event = None
-            while True:
-                with self._timestamp_lock:
-                    if (
-                        self._timeout_timestamp is not None
-                        and datetime.now() > self._timeout_timestamp
-                    ):
-                        self._timeout_timestamp = None
-                        break
+            while (
```

Comment:
I don't think we need this anymore.
```
self._timeout_timestamp is None or datetime.now() <= self._timeout_timestamp
```
This was due to the sync nature of the event.py

Reply:
OK, clarified via discussion.